src/playback/internal/playbackcontroller.cpp

```diff
@@ -1631,3 +1635,15 @@ muse::audio::secs_t PlaybackController::playedTickToSecs(int tick) const
 {
     return secs_t(notationPlayback()->playedTickToSec(tick));
 }
+
+void PlaybackController::toggleHearPlaybackWhenEditing()
+{
+    // Just to check if the function has been called from noteinputpreferencesmodel or from playbackcontroller
+    // If the configuration() stored value is the same as the local value we have been called from playbackcontroller to change the value
+    if (m_hearPlaybackWhenEditing == configuration()->playNotesWhenEditing()) {
+        configuration()->setPlayNotesWhenEditing(!m_hearPlaybackWhenEditing);
+    }
+    m_hearPlaybackWhenEditing = !m_hearPlaybackWhenEditing;
+
+    notifyActionCheckedChanged(TOGGLE_HEAR_PLAYBACK_WHEN_EDITING);
```

Comment:
And to make this work properly when the value is changed via Preferences, don't add this here; instead, add a `playNotesWhenEditingChanged` `Notification` to `IPlaybackConfiguration` (a good example is `NotationConfiguration::isPlayChordSymbolsChanged`), and then in `PlaybackController::init`, subscribe to this `playNotesWhenEditingChanged` notification, with a callback that calls `notifyActionCheckedChanged(TOGGLE_HEAR_PLAYBACK_WHEN_EDITING);`. 

Alternatively, make this subscription directly in `PlaybackUiActions::init()`, with a callback that calls `m_actionCheckedChanged.send({ "toggle-hear-playback-when-editing" });`.